pkg/db/driver_mysql.go

```diff
@@ -24,25 +23,21 @@ func NewMysqlDriverFactory() DriverFactory {
 type mysqlDriverFactory struct{}
 
 func (m *mysqlDriverFactory) GetDSN(settings Settings) string {
-	dsn := url.URL{
-		User: url.UserPassword(settings.Uri.User, settings.Uri.Password),
-		Host: fmt.Sprintf("tcp(%s:%d)", settings.Uri.Host, settings.Uri.Port),
-		Path: settings.Uri.Database,
-	}
-
-	qry := dsn.Query()
-	qry.Set("multiStatements", "true")
-	qry.Set("parseTime", strconv.FormatBool(settings.ParseTime))
-	qry.Set("charset", "utf8mb4")
-	dsn.RawQuery = qry.Encode()
-
-	uri := dsn.String()
-
-	return uri[2:]
+	cfg := mysql.NewConfig()
+	cfg.User = settings.Uri.User
+	cfg.Passwd = settings.Uri.Password
+	cfg.Net = "tcp"
+	cfg.Addr = net.JoinHostPort(settings.Uri.Host, strconv.Itoa(settings.Uri.Port))
+	cfg.DBName = settings.Uri.Database
+	cfg.MultiStatements = true
+	cfg.ParseTime = true
+	cfg.Collation = "utf8mb4_general_ci"
```

Comment:
should we make this part of the settings too, so the user can configure and overwrite it?

Reply:
Also we have been setting a charset before, now we are setting a collation. According to the release notes of go-sql-driver/mysql 1.8.0 we are staying compatible when not specifying a Collation or Charset at all.